phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2639,6 +2646,20 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
                         QueryServices.ALLOW_SPLITTABLE_SYSTEM_CATALOG_ROLLBACK);
                 }
             }
+            if (settingNewPhoenixTTLAttribute(tableMetadata, PHOENIX_TTL_BYTES)) {
+                // Disallow if the parent has PHOENIX_TTL set.
+                if (parentTable != null &&  parentTable.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED) {
+                    isSchemaMutationAllowed = false;
+                }
+
+                // Since we do not allow propagation of PHOENIX_TTL values during ALTER for now.
+                // If a child view exists and the parent previously had a PHOENIX_TTL value set then that
```

Comment:
We aren't checking the "and the parent previously had a PHOENIX_TTL value set " part. Shouldn't we do that too?

Reply:
As discussed offline, changing the comment will bring more clarity. The checks are for "this" view's parent and its children